### PR TITLE
Ensures flutter jar is added to all build types on plugin projects

### PR DESCRIPTION
**Pull request by josh-burton**
## Description

The flutter jar is currently only added to debug/profile/release build types for plugin projects.

Plugin projects could have extra build types, or an app could add extra build types to all subprojects:

```
subprojects {
	android.buildTypes {
		debug {}
        staging {}
        release {}
	}
}

```
This causes compilation failures as the plugin projects can't find Flutter classes.

## Related Issues

#34089

## Tests

Added a test to gradle_plugin_light_apk_test.dart which tests building an android app containing custom build types in all subprojects (i.e. plugins)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

@blasten Requesting your review on this one :)

### CODE REVIEW

**Comment by athornz** [comment]
This may require some rework once #34592 is merged to keep the logic consistent.

**Comment by athornz** [comment]
@blasten this PR is ready to go now, would you mind taking a look?

**Comment by athornz** [comment]
@dnfield @matthew-carroll who is the best person to review this?

**Comment by matthew-carroll** [comment]
Probably @blasten

File: dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart [hunk 1]
```diff
@@ -159,53 +159,14 @@ Future<void> main() async {
       });
 
       await runProjectTest((FlutterProject project) async {
-        section('gradlew assembleBeta (custom release build)');
-        await project.addCustomBuildType('beta', initWith: 'release');
```
Comment (blasten) [inline dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart]: why removing these tests?

Comment (athornz) [reply dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart]: Oops, must have been a merge error!

**Comment by athornz** [comment]
Thanks for the review @blasten, I've addressed all of the comments.

**Review by blasten — APPROVED** [review]
Thanks for cleaning this up!

**Comment by blasten** [comment]
It looks like this breaks one of the smoke tests.

**Comment by dnfield** [comment]
I think that test is failing because of an unrelated issue - it should be safe to ignore, and should pass on post submit.

**Comment by dnfield** [comment]
(The contributor isn't able to access the encrypted variable)

**Comment by blasten** [comment]
Gotcha. Thanks @dnfield!

**Comment by dnfield** [comment]
The test passed on post submit. We should probably improve this test so that it doesn't fail if the JSON is invalid for the service account, or at least somehow make the test itself make it clear that it failed but is safe to ignore.

**Pull request merged**